homeassistant/components/geonetnz_quakes/geo_location.py

```diff
@@ -145,6 +155,12 @@ def get_entry(self, external_id):
         """Remove entity."""
         async_dispatcher_send(self._hass, SIGNAL_DELETE_ENTITY.format(external_id))
 
+    async def _status_update(self, status_info):
+        """Propagate status update."""
+        _LOGGER.debug("Status update received: %s", status_info)
+        self._status_info = status_info
+        async_dispatcher_send(self._hass, SIGNAL_STATUS.format(self._config_entry_id))
```

Comment:
It's not good that we depend on the geolocation platform to update the sensor platform. We should probably refactor to move the manager to the component, and have a separate process for adding entities in the platforms. We can signal the platforms from the component for updates.

Reply:
OK. I moved the manager to the component. The sensor platform remains almost unchanged, except for passing in the manager from the setup method. I had to make more major changes to the geolocation platform: It is waiting for dispatcher events instructing it to create a new entity. The manager now needs to wait for its first update until the component is loaded, and the geolocation platform has actually subscribed to dispatcher events.
I added a few more debug log statement along the way to ensure that everything is still working as expected.